src/Sarif/Writers/SarifLogger.cs

```diff
@@ -466,14 +456,14 @@ public bool ShouldLog(ResultLevel level)
 
         public void LogToolNotification(Notification notification)
         {
-            _run.ToolNotifications = _run.ToolNotifications ?? new List<Notification>();
-            _run.ToolNotifications.Add(notification);
+            _run.Invocation.ToolNotifications = _run.Invocation.ToolNotifications ?? new List<Notification>();
```

Comment:
Create `Invocation` if it doesn't exist? Or are we sure it exists at this point?